src/harness/core/comparers.ts

```diff
@@ -0,0 +1,43 @@
+namespace core {
+    export function compareNumbers(a: number, b: number): number {
+        if (a === b) return 0;
+        if (a === undefined) return -1;
+        if (b === undefined) return +1;
+        return a < b ? -1 : +1;
+    }
+
+    export function compareStrings(a: string, b: string, ignoreCase: boolean): number {

```

Comment:
how are these diffrent from the ones in src\compiler\core?

Reply:
They aren't. Originally I intended to attempt to divorce the harness from the compiler to improve time-to-test by reducing compilation time, but with @RyanCavanaugh's forthcoming changes for project references this is no longer an issue. I am in the process of switching over to using existing functions, which should show up in the next few commits.